File: app/src/main/java/studio/mandysa/music/service/model/DefaultAlbum.java
package studio.mandysa.music.service.model;

import java.util.ArrayList;

public class DefaultAlbum implements AlbumModel {

    private final ArrayList<ArtistModel> mArtistModel = new ArrayList<>();

    private final ArrayList<MusicModel> mMusicList = new ArrayList<>();

    @Override
    public void add(MusicModel musicModel) {
        mMusicList.add(musicModel);
    }

    @Override
    public void remove(MusicModel musicModel) {
        mMusicList.remove(musicModel);
    }

    @Override
    public int size() {
        return mMusicList.size();
    }

    @Override
    public MusicModel get(int index) {
        return mMusicList.get(index);
    }

    @Override
    public ArrayList<ArtistModel> getArtist() {
        return mArtistModel;
    }

}

File: media/src/main/java/mandysax/media/model/DefaultAlbum.kt
package mandysax.media.model

import java.util.*

/**
 * @author liuxiaoliu66
 */
class DefaultAlbum : AlbumModel<DefaultMusic<DefaultArtist>> {
    override val artist = ArrayList<DefaultArtist>()
    private val mMusicList = ArrayList<DefaultMusic<DefaultArtist>>()

    @SafeVarargs
    override fun add(vararg musicModels: DefaultMusic<DefaultArtist>) {
        mMusicList.addAll(listOf(*musicModels))
    }

    @SafeVarargs
    override fun remove(vararg musicModels: DefaultMusic<DefaultArtist>) {
        mMusicList.removeAll(listOf(*musicModels))
    }

    override fun size(): Int {
        return mMusicList.size
    }

    override fun get(index: Int): DefaultMusic<DefaultArtist> {
        return mMusicList[index]
    }
}